dynamic_programming/longest_common_subsequence.cpp

```diff
@@ -0,0 +1,303 @@
+/**
+ * @file
+ * @author [Santosh Jonnakuti](https://github.com/Santoshjonnakuti)
+ * @brief [Longest Common Subsequence Algorithm](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem)
+ * implementation in C to find the LCS between 2 strings.
+ * @details LCS algorithm uses dynamic programming approach to find all possible LCS between 2 strings. 
+ * Dynamic Programming follow a method so that even sub-problems will also have optimal solution.
+ *
+ * LCS() will find the length of the LCS of the following two strings 'GAC' and 'AGCAT'.
+ * LCS function will return the following 2D-Matrix
+ *
+ *      0 0 0 0 0 0
+ *      0 0 1 1 1 1
+ *      0 1 1 1 2 2
+ *      0 1 1 2 2 2
+ *
+ * So Length of LCS will be 2.
+ *
+ * allPossibleLSC() will print all possible LCS between those strings.
+ *
+ * AC
+ * GC
+ * GA
+ * GA
+ * 
+ * To test run './longest_common_subsequence -test'
+ * So There are 4 Longest common subsequences in 'GAC' and 'AGCAT'
+ */
+
+
+#include <stdio.h>      /// for I/O operations
+#include <stdlib.h>     /// for dynamic memory allocation
+#include <string.h>     /// for `puts()`, `gets()`, and `strlen()`
+#include <assert.h>     /// for `assert()`
+
+#define MAX 100      // maximum size of string1 and string2
+
+/**
+ * @brief Structure to store all LCS and no of possible LCS
+ */
+struct array
+{
+	char **LCSArray;      // array to store all possible LCS
+	int length;           // no of possible LCs
+} variable;
+
+
+// Functions Declaration
+int max(int a, int b);
+int** LCS(char *string1, int m, char *string2, int n);
+void allPossibleLCS(int **c, char *a, int index, int i, int j, char *string1, char *string2, int lengthOfLCS);
+void findAllLCS(char *string1, char *string2);
+static void test();
```

Comment:
I don't think this is needed. It could cause a bit more confusion to the reader IMO. Although it's not required to remove them, it's better to simply declare the function without its prototype.